asset/src/routed_sender/processor.ts

```diff
@@ -0,0 +1,122 @@
+            const keys = keyset.split(',');
+
+            for (const key of keys) {
+                this.routeDict.set(key, config);
+            }
+        }
+    }
+
+    private async createRoute(route: string) {
+        const { api_name } = this.opConfig;
+        const config = this.routeDict.get(route);
+        // we cannot cache this type of api becuase we need to call and get several types
+        // @ts-expect-error
+        const client: RouteSenderAPI = await this.context.apis.executionContext._apis[api_name]
+            .instance.createAPI(config);
+        this.routingExectuion.set(route, { client, data: [] });
+        await client.verifyRoute(config);
+    }
+
+    private sendRecords(client: RouteSenderAPI, data: DataEntity[]) {
+        if (data.length === 0) return [];
+        return chunk(data, this.limit)
+            .map((chunkedData) => client.send(chunkedData));
+    }
+
+    async routeToDestinations(batch: DataEntity[]): Promise<void> {
+        for (const record of batch) {
+            const route = record.getMetadata('standard:route');
+            // if we have route, then use it, else make a topic if allowed.
+            // if not then check if a "*" is set, if not then use rejectRecord
+            if (this.routingExectuion.has(route)) {
+                const routeConfig = this.routingExectuion.get(route) as Endpoint;
+                routeConfig.data.push(record);
+            } else if (this.routeDict.has(route)) {
+                await this.createRoute(route);
+
+                const routeConfig = this.routingExectuion.get(route) as Endpoint;
+                routeConfig.data.push(record);
+            } else if (this.routeDict.has('*')) {
+                if (!this.routingExectuion.has('*')) {
+                    await this.createRoute('*');
+                }
+                const routeConfig = this.routingExectuion.get('*') as Endpoint;
+                routeConfig.data.push(record);
+            } else if (this.routeDict.has('**')) {
+                if (!this.routingExectuion.has('**')) {
+                    await this.createRoute('**');
+                }
+
+                const routeConfig = this.routingExectuion.get('**') as Endpoint;
+                routeConfig.data.push(record);
+            } else {
+                let error: TSError;
+
+                if (route == null) {
+                    error = new TSError('No route was specified in record metadata');
+                } else {
+                    error = new TSError(`Invalid connection route: ${route} was not found on connector_map`);
+                }
+
+                this.rejectRecord(record, error);
+            }
+        }
+
+        const senders = [];
+
+        for (const [, { data, client }] of this.routingExectuion) {
+            if (data.length > 0) {
+                senders.push(...this.sendRecords(client, data));
+            }
+        }
+
+        await Promise.all(senders);
```

Comment:
This should probably use `pMap` with some sort of concurrency control or maybe it should run serially?